File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnRhythmStationOptions.java

package com.gracenote.gnsdk;

/** 
* Configures options for {@link GnRhythmStation} 
*/ 
 
public class GnRhythmStationOptions extends GnRhythmQueryOptions {
  private transient long swigCPtr;

  protected GnRhythmStationOptions(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnRhythmStationOptions_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnRhythmStationOptions obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnRhythmStationOptions(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public void custom(String option, String value) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnRhythmStationOptions_custom(swigCPtr, this, option, value);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnRhythmStationOptions.cs

namespace GracenoteSDK {

/**
* Configures options for GnRhythmStation
*/
public class GnRhythmStationOptions : GnRhythmQueryOptions {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnRhythmStationOptions(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnRhythmStationOptions_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnRhythmStationOptions obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnRhythmStationOptions() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnRhythmStationOptions(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public override void Custom(string option, string value) {
    gnsdk_csharp_marshalPINVOKE.GnRhythmStationOptions_Custom(swigCPtr, option, value);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

}

}
